### PR TITLE
Add: universal Card component

### PR DESCRIPTION
**Pull request by vincit-matu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

Adds a `Card` component to common components, which displays a plain card with `heading` and `text`.
It can also display:
  - an image (if given an `imageSrc). The image uses `heading`as `alt` text by default, but `imageAlt` allows the use of a any value.
  - `infos` (a `value` with an optional `icon` and `testId`)
  - `tags` ((== `JSX.Element`s, like `StatusLabel`s and `Tag`s)
  - `buttons` (== `JSX.Element`s, like `Button`s, `ButtonLikeLink`s etc)
  - child elements, for the moment with the same positioning as `infos`, as our use-cases are mutually exclusive between the two
- if given a `link`, it is used with the image and heading text
- any child elements will be displayed after the main `text`

Use `margin` to apply one on the entire card, `backgroundColor` for a custom background color, and `testId`/`headingTestId`/`textTestId` to provide `data-test-id`s for each element respectively. To use `data-test-id`s for `tags` and `buttons` they should be included in JSX.Elements you provide)

The replaced cards:
 - Single reservation: search result card and my reservations listing cards
 - Recurring reservations: application round card and my applications cards
 - Reservation unit page: related units cards, on the bottom of the screen
 - Admin pages reservation unit card, on a unit's page

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Manual testing of the different cards in different screen sizes.

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- TILA-3522

### CODE REVIEW

File: apps/admin-ui/src/spa/unit/[id]/ReservationUnitCard.tsx [hunk 1]
```diff
@@ -113,58 +38,64 @@ export function ReservationUnitCard({
...
+      ]
+    : [
+        <StatusLabel
+          type="success"
+          icon={<IconCheck ariaHidden />}
+          key="success"
+        >
+          {t("ReservationUnitCard.statePublished")}
+        </StatusLabel>,
+      ];
```
Comment (vincit-matu) [inline apps/admin-ui/src/spa/unit/[id]/ReservationUnitCard.tsx:77]: useMemo?

File: apps/ui/components/reservation-unit/RelatedUnits.tsx [hunk 2]
```diff
@@ -30,184 +29,110 @@ type PropsType = {
...
+              <IconTicket aria-label={t("prices:reservationUnitPriceLabel")} />
+            ),
+            value: unitPrice,
+          });
+        }
+        const buttons = [
+          <ButtonLikeLink
+            href={reservationUnitPath(unit.pk ?? 0)}
+            key={t("reservationUnitCard:seeMore")}
+            onClick={() => {
```
Comment (vincit-matu) [inline apps/ui/components/reservation-unit/RelatedUnits.tsx:112]: unnucessary onClick

File: packages/common/src/components/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+  order: -1;
+  justify-content: flex-start;
+  margin-bottom: var(--spacing-2-xs);
+
+  /* display the tags on top of the image */
+  .card--with-image & {
+    margin-top: calc((var(--spacing-layout-m) + var(--spacing-2-xs)) * -1);
+  }
+
+  /* TODO: This shouldn't be necessary if using the updated custom StatusLabel as it is the correct height. Check to make sure, and remove the "> *" declarations */
```
Comment (joonatank) [inline packages/common/src/components/Card.tsx:249]: still?

**Review by joonatank — APPROVED** [review]
Fix the `h6` allowing overriding with the correct heading level (per page), otherwise 👍

File: packages/common/src/components/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+    }
+  }
+`;
+
+const Text = styled.p`
+  font-size: var(--fontsize-body-m);
+  margin: 0;
+`;
+
+const Header = styled.h6`
```
[the remark this answers is not in the document]
Comment (vincit-matu) [reply packages/common/src/components/Card.tsx:311]: The heading level is now dynamic. A bit hacky perhaps (considering the typecasting to ElementType), but it seems to work.

**Pull request merged**